enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -99,259 +116,211 @@ This is often required for industries such as the US public sector, healthcare o
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
 
-#### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
+* Server-authenticated TLS is enabled if `destination` is a secure URL (e.g. 'https:')
+* Client-authenticated TLS is enabled if `destination` is secure *and* `secretRef` has keys `tls.crt`, `tls.key`, `ca.crt`
+  - it is an error if `secretRef` is present but `destination` is *not* secure, or the required 'tls.' keys are missing or invalid.
+* An intentionally insecure output (no TLS) must have `insecure: true`
+  - avoid accidental insecure configuration of an output that was intended to be secure.
+* The user is responsible for creating and maintaining the secret objects
+* The cluster logging operator is responsible for watching secrets and applying changes,  e.g. if the user replaces a certificate or changes a password in a secret, the operator must re-connect affected outputs with the new credentials.
+
+#### Reliability
+* Output connections will automatically re-connect on disconnect.
+* Changes to secrets will trigger automatic re-connect with new credentials.
+
+#### Scale
+* The cluster logging operator generates configuration for the collector
+  - with a singleton ClusterLogForwarder this is unlikely to be a scaling problem.
+* The actual connection and forwarding to remote endpoints is done by the collector
+  - we rely on the collector to scale and perform.
+
+#### Metrics
+
+Note: the actual forwarding is done by the collector, so we can only provide metrics that are available from the collector.
+
+Relevant metrics include:
+
+* Counter:
+  - Volume (bytes) per input, per pipeline, per output and total.
+* Histogram/Summary:
+  - Throughput (bytes/sec) per input, per pipeline, per output and total.
+  - Read size: per input
+  - Write size: per output
+  - Latency (sec)
+    - per pipeline: from read to written on all outputs.
+    - per output: from read to written on this output.
+
+#### Cluster Logging Operator
+
+The `cluster-logging-operator` will use the `ClusterLogForwarder` configuration to:
+
+* generate output configuration for the collector that respects all the pipelines.
+* mount secrets in the collector daemonset as needed for each endpoint.
+  - `/var/run/ocp-collector/secrets/<secretName>/{tls.crt,tls.key,ca-bundle.crt,username,password}`
+    * `secretName` is the name of a secret.
+    * `tls.crt`, `tls.key`, `ca-bundle.crt` are files in the secret
```

Comment:
`ca-bundle.crt` vs `ca.crt` ?

Reply:
Done. Things I've worked with in the past used "CA cert chain" or just "CA", but on googling I see that "bundle" is also a widespread term, so I'll put it back.